sdks/python/apache_beam/transforms/combiners_test.py

```diff
@@ -31,24 +31,43 @@
 import apache_beam as beam
 import apache_beam.transforms.combiners as combine
 from apache_beam.options.pipeline_options import PipelineOptions
+from apache_beam.options.pipeline_options import StandardOptions
 from apache_beam.testing.test_pipeline import TestPipeline
 from apache_beam.testing.test_stream import TestStream
 from apache_beam.testing.util import assert_that
 from apache_beam.testing.util import equal_to
 from apache_beam.testing.util import equal_to_per_window
+from apache_beam.transforms import trigger
 from apache_beam.transforms import window
 from apache_beam.transforms.core import CombineGlobally
 from apache_beam.transforms.core import Create
 from apache_beam.transforms.core import Map
 from apache_beam.transforms.display import DisplayData
 from apache_beam.transforms.display_test import DisplayDataItemMatcher
 from apache_beam.transforms.ptransform import PTransform
+from apache_beam.transforms.trigger import AfterAll
+from apache_beam.transforms.trigger import AfterCount
+from apache_beam.transforms.trigger import AfterWatermark
+from apache_beam.transforms.window import GlobalWindows
 from apache_beam.transforms.window import TimestampCombiner
 from apache_beam.typehints import TypeCheckError
 from apache_beam.utils.timestamp import Timestamp
 
 
 class CombineTest(unittest.TestCase):
+  def setUp(self):
+    # Use state on the TestCase class, since other references would be pickled
+    # into a closure and not have the desired side effects.
+    CombineTest.all_records = []
```

Comment:
This make it so we can't make any of these validates runner. If you need to record things in order, use state.

Reply:
When you said "state", did you mean a class attribute or the StateSpec of Beam?

I tried other ways to record firing panes, but after synced with the newest version of apache_beam, it turns out the result is no longer the last firing, but a list of all firings. Therefore there is no need to record anything. I also tried the DISCARDING mode, its result also correctly represented the firings ([1, 2, 3, 4, 5, 0, 0]).